In [1]:
import pandas as pd

# Set the path to your datasets folder
path = r"E:\NILE TECH\DataAnalyst90Days\Phase1_Olist\datasets\\"

# Load all 9 CSV files
orders           = pd.read_csv(path + "olist_orders_dataset.csv")
order_items      = pd.read_csv(path + "olist_order_items_dataset.csv")
order_payments   = pd.read_csv(path + "olist_order_payments_dataset.csv")
order_reviews    = pd.read_csv(path + "olist_order_reviews_dataset.csv")
customers        = pd.read_csv(path + "olist_customers_dataset.csv")
sellers          = pd.read_csv(path + "olist_sellers_dataset.csv")
products         = pd.read_csv(path + "olist_products_dataset.csv")
geolocation      = pd.read_csv(path + "olist_geolocation_dataset.csv")
translations     = pd.read_csv(path + "product_category_name_translation.csv")

print("All 9 files loaded successfully!")

All 9 files loaded successfully!


In [2]:
print("=== ORDERS ===")
print("Shape:", orders.shape)          # rows and columns
print("\nColumn names:")
print(orders.columns.tolist())         # all column names
print("\nFirst 3 rows:")
orders.head(3)   

=== ORDERS ===
Shape: (99441, 8)

Column names:
['order_id', 'customer_id', 'order_status', 'order_purchase_timestamp', 'order_approved_at', 'order_delivered_carrier_date', 'order_delivered_customer_date', 'order_estimated_delivery_date']

First 3 rows:


,order_id,customer_id,order_status,order_purchase_timestamp,order_approved_at,order_delivered_carrier_date,order_delivered_customer_date,order_estimated_delivery_date
0,e481f51cbdc54678b7cc49136f2d6af7,9ef432eb6251297304e76186b10a928d,delivered,2017-10-02 10:56:33,2017-10-02 11:07:15,2017-10-04 19:55:00,2017-10-10 21:25:13,2017-10-18 00:00:00
1,53cdb2fc8bc7dce0b6741e2150273451,b0830fb4747a6c6d20dea0b8c802d7ef,delivered,2018-07-24 20:41:37,2018-07-26 03:24:27,2018-07-26 14:31:00,2018-08-07 15:27:45,2018-08-13 00:00:00
2,47770eb9100c2d0c44946d9cf07ec65d,41ce2a54c0b03bf3443c3d931a367089,delivered,2018-08-08 08:38:49,2018-08-08 08:55:23,2018-08-08 13:50:00,2018-08-17 18:06:29,2018-09-04 00:00:00


## Orders Table
The master table — every order ever placed on the Olist marketplace.
One row = one customer order.

- **Rows:** ~99,441 | **Columns:** 8
- **Date range:** September 2016 to October 2018
- **Key column:** `order_id` — the primary key that links to all other tables
- **Status types:** delivered, shipped, cancelled, invoiced, unavailable, processing, created

**Data quality note:** `order_delivered_customer_date` has ~2,965 nulls
— these are orders that were never delivered (cancelled or still in transit).

In [4]:
print("=== ORDERS ITEMS ===")
print("Shape:", order_items.shape)          # rows and columns
print("\nColumn names:")
print(order_items.columns.tolist())         # all column names
print("\nFirst 3 rows:")
orders.head(3)   

=== ORDERS ITEMS ===
Shape: (112650, 7)

Column names:
['order_id', 'order_item_id', 'product_id', 'seller_id', 'shipping_limit_date', 'price', 'freight_value']

First 3 rows:


,order_id,customer_id,order_status,order_purchase_timestamp,order_approved_at,order_delivered_carrier_date,order_delivered_customer_date,order_estimated_delivery_date
0,e481f51cbdc54678b7cc49136f2d6af7,9ef432eb6251297304e76186b10a928d,delivered,2017-10-02 10:56:33,2017-10-02 11:07:15,2017-10-04 19:55:00,2017-10-10 21:25:13,2017-10-18 00:00:00
1,53cdb2fc8bc7dce0b6741e2150273451,b0830fb4747a6c6d20dea0b8c802d7ef,delivered,2018-07-24 20:41:37,2018-07-26 03:24:27,2018-07-26 14:31:00,2018-08-07 15:27:45,2018-08-13 00:00:00
2,47770eb9100c2d0c44946d9cf07ec65d,41ce2a54c0b03bf3443c3d931a367089,delivered,2018-08-08 08:38:49,2018-08-08 08:55:23,2018-08-08 13:50:00,2018-08-17 18:06:29,2018-09-04 00:00:00


## Order Items Table
One row = one item within an order. An order can have multiple items,
which is why this table has more rows than the orders table.

- **Rows:** ~112,650 | **Columns:** 7
- **Key columns:** `order_id` (links to orders), `product_id` (links to products), `seller_id` (links to sellers)
- **Price column:** the item price in Brazilian Reals (R$)
- **freight_value:** shipping cost for that item

**Insight:** More rows than orders because many customers buy multiple items
in a single order. Total order value = sum of price + freight_value per order_id.

In [5]:
print("=== ORDERS PAYMENTS ===")
print("Shape:", order_payments.shape)          # rows and columns
print("\nColumn names:")
print(order_payments.columns.tolist())         # all column names
print("\nFirst 3 rows:")
orders.head(3)

=== ORDERS PAYMENTS ===
Shape: (103886, 5)

Column names:
['order_id', 'payment_sequential', 'payment_type', 'payment_installments', 'payment_value']

First 3 rows:


,order_id,customer_id,order_status,order_purchase_timestamp,order_approved_at,order_delivered_carrier_date,order_delivered_customer_date,order_estimated_delivery_date
0,e481f51cbdc54678b7cc49136f2d6af7,9ef432eb6251297304e76186b10a928d,delivered,2017-10-02 10:56:33,2017-10-02 11:07:15,2017-10-04 19:55:00,2017-10-10 21:25:13,2017-10-18 00:00:00
1,53cdb2fc8bc7dce0b6741e2150273451,b0830fb4747a6c6d20dea0b8c802d7ef,delivered,2018-07-24 20:41:37,2018-07-26 03:24:27,2018-07-26 14:31:00,2018-08-07 15:27:45,2018-08-13 00:00:00
2,47770eb9100c2d0c44946d9cf07ec65d,41ce2a54c0b03bf3443c3d931a367089,delivered,2018-08-08 08:38:49,2018-08-08 08:55:23,2018-08-08 13:50:00,2018-08-17 18:06:29,2018-09-04 00:00:00


## Order Payments Table
How each order was paid for. One order can have multiple payment rows
(e.g. part credit card, part voucher).

- **Rows:** ~103,886 | **Columns:** 5
- **Payment types:** credit_card, boleto, voucher, debit_card
- **payment_installments:** how many monthly instalments the customer chose
- **payment_value:** total amount paid for that payment method

**Insight:** Brazil's most common payment method is boleto (a bank slip).
Credit card users tend to pay in multiple instalments — common in Brazilian retail.

In [6]:
print("=== ORDERS REVIEWS ===")
print("Shape:", order_reviews.shape)          # rows and columns
print("\nColumn names:")
print(order_reviews.columns.tolist())         # all column names
print("\nFirst 3 rows:")
orders.head(3)

=== ORDERS REVIEWS ===
Shape: (99224, 7)

Column names:
['review_id', 'order_id', 'review_score', 'review_comment_title', 'review_comment_message', 'review_creation_date', 'review_answer_timestamp']

First 3 rows:


,order_id,customer_id,order_status,order_purchase_timestamp,order_approved_at,order_delivered_carrier_date,order_delivered_customer_date,order_estimated_delivery_date
0,e481f51cbdc54678b7cc49136f2d6af7,9ef432eb6251297304e76186b10a928d,delivered,2017-10-02 10:56:33,2017-10-02 11:07:15,2017-10-04 19:55:00,2017-10-10 21:25:13,2017-10-18 00:00:00
1,53cdb2fc8bc7dce0b6741e2150273451,b0830fb4747a6c6d20dea0b8c802d7ef,delivered,2018-07-24 20:41:37,2018-07-26 03:24:27,2018-07-26 14:31:00,2018-08-07 15:27:45,2018-08-13 00:00:00
2,47770eb9100c2d0c44946d9cf07ec65d,41ce2a54c0b03bf3443c3d931a367089,delivered,2018-08-08 08:38:49,2018-08-08 08:55:23,2018-08-08 13:50:00,2018-08-17 18:06:29,2018-09-04 00:00:00


## Order Reviews Table
Customer satisfaction data — star ratings (1-5) and written comments
left after delivery.

- **Rows:** ~99,224 | **Columns:** 7
- **review_score:** 1 (worst) to 5 (best) — our main satisfaction metric
- **review_comment_message:** written review — has ~58,247 nulls (most customers only leave a score)
- **review_comment_title:** short title — has ~87,656 nulls

**Business question this enables:** Does late delivery cause lower review scores?
This table + delivery dates from orders will answer that in Week 2.

In [7]:
print("=== CUSTOMERS ===")
print("Shape:", customers.shape)          # rows and columns
print("\nColumn names:")
print(customers.columns.tolist())         # all column names
print("\nFirst 3 rows:")
orders.head(3)

=== CUSTOMERS ===
Shape: (99441, 5)

Column names:
['customer_id', 'customer_unique_id', 'customer_zip_code_prefix', 'customer_city', 'customer_state']

First 3 rows:


,order_id,customer_id,order_status,order_purchase_timestamp,order_approved_at,order_delivered_carrier_date,order_delivered_customer_date,order_estimated_delivery_date
0,e481f51cbdc54678b7cc49136f2d6af7,9ef432eb6251297304e76186b10a928d,delivered,2017-10-02 10:56:33,2017-10-02 11:07:15,2017-10-04 19:55:00,2017-10-10 21:25:13,2017-10-18 00:00:00
1,53cdb2fc8bc7dce0b6741e2150273451,b0830fb4747a6c6d20dea0b8c802d7ef,delivered,2018-07-24 20:41:37,2018-07-26 03:24:27,2018-07-26 14:31:00,2018-08-07 15:27:45,2018-08-13 00:00:00
2,47770eb9100c2d0c44946d9cf07ec65d,41ce2a54c0b03bf3443c3d931a367089,delivered,2018-08-08 08:38:49,2018-08-08 08:55:23,2018-08-08 13:50:00,2018-08-17 18:06:29,2018-09-04 00:00:00


## Customers Table
Geographic information about each customer. One row = one unique customer.

- **Rows:** ~99,441 | **Columns:** 5
- **customer_unique_id:** the true unique customer ID (a customer can place multiple orders)
- **customer_city / customer_state:** where the customer lives
- **customer_zip_code_prefix:** used to join with geolocation table for mapping

**Note:** `customer_id` in this table is order-specific (changes per order).
`customer_unique_id` is the real repeat-customer identifier — use this for retention analysis.

In [8]:
print("=== SELLERS ===")
print("Shape:", sellers.shape)          # rows and columns
print("\nColumn names:")
print(sellers.columns.tolist())         # all column names
print("\nFirst 3 rows:")
orders.head(3)

=== SELLERS ===
Shape: (3095, 4)

Column names:
['seller_id', 'seller_zip_code_prefix', 'seller_city', 'seller_state']

First 3 rows:


,order_id,customer_id,order_status,order_purchase_timestamp,order_approved_at,order_delivered_carrier_date,order_delivered_customer_date,order_estimated_delivery_date
0,e481f51cbdc54678b7cc49136f2d6af7,9ef432eb6251297304e76186b10a928d,delivered,2017-10-02 10:56:33,2017-10-02 11:07:15,2017-10-04 19:55:00,2017-10-10 21:25:13,2017-10-18 00:00:00
1,53cdb2fc8bc7dce0b6741e2150273451,b0830fb4747a6c6d20dea0b8c802d7ef,delivered,2018-07-24 20:41:37,2018-07-26 03:24:27,2018-07-26 14:31:00,2018-08-07 15:27:45,2018-08-13 00:00:00
2,47770eb9100c2d0c44946d9cf07ec65d,41ce2a54c0b03bf3443c3d931a367089,delivered,2018-08-08 08:38:49,2018-08-08 08:55:23,2018-08-08 13:50:00,2018-08-17 18:06:29,2018-09-04 00:00:00


## Sellers Table
Information about the sellers on the Olist marketplace.
One row = one unique seller.

- **Rows:** ~3,095 | **Columns:** 4
- **seller_id:** links to order_items — tells us which seller fulfilled each order item
- **seller_city / seller_state:** where the seller is located in Brazil

**Business question this enables:** Which sellers have the worst delivery times?
Which states have the best-performing sellers?
seller_id links order_items → sellers for this analysis.

In [9]:
print("=== PRODUCTS ===")
print("Shape:", products.shape)          # rows and columns
print("\nColumn names:")
print(products.columns.tolist())         # all column names
print("\nFirst 3 rows:")
orders.head(3)

=== PRODUCTS ===
Shape: (32951, 9)

Column names:
['product_id', 'product_category_name', 'product_name_lenght', 'product_description_lenght', 'product_photos_qty', 'product_weight_g', 'product_length_cm', 'product_height_cm', 'product_width_cm']

First 3 rows:


,order_id,customer_id,order_status,order_purchase_timestamp,order_approved_at,order_delivered_carrier_date,order_delivered_customer_date,order_estimated_delivery_date
0,e481f51cbdc54678b7cc49136f2d6af7,9ef432eb6251297304e76186b10a928d,delivered,2017-10-02 10:56:33,2017-10-02 11:07:15,2017-10-04 19:55:00,2017-10-10 21:25:13,2017-10-18 00:00:00
1,53cdb2fc8bc7dce0b6741e2150273451,b0830fb4747a6c6d20dea0b8c802d7ef,delivered,2018-07-24 20:41:37,2018-07-26 03:24:27,2018-07-26 14:31:00,2018-08-07 15:27:45,2018-08-13 00:00:00
2,47770eb9100c2d0c44946d9cf07ec65d,41ce2a54c0b03bf3443c3d931a367089,delivered,2018-08-08 08:38:49,2018-08-08 08:55:23,2018-08-08 13:50:00,2018-08-17 18:06:29,2018-09-04 00:00:00


## Products Table
Details about each product listed on Olist.
One row = one unique product.

- **Rows:** ~32,951 | **Columns:** 9
- **product_id:** links to order_items — this is the key join column
- **product_category_name:** in Portuguese — needs translation file to read in English
- **product_weight_g / dimensions:** physical size — affects freight costs

**Data quality note:** Some products have nulls in weight and dimension columns.
Category names are in Portuguese — join with `translations` table to get English names.

In [10]:
print("=== GEOLOCATION ===")
print("Shape:", geolocation.shape)          # rows and columns
print("\nColumn names:")
print(geolocation.columns.tolist())         # all column names
print("\nFirst 3 rows:")
orders.head(3)

=== GEOLOCATION ===
Shape: (1000163, 5)

Column names:
['geolocation_zip_code_prefix', 'geolocation_lat', 'geolocation_lng', 'geolocation_city', 'geolocation_state']

First 3 rows:


,order_id,customer_id,order_status,order_purchase_timestamp,order_approved_at,order_delivered_carrier_date,order_delivered_customer_date,order_estimated_delivery_date
0,e481f51cbdc54678b7cc49136f2d6af7,9ef432eb6251297304e76186b10a928d,delivered,2017-10-02 10:56:33,2017-10-02 11:07:15,2017-10-04 19:55:00,2017-10-10 21:25:13,2017-10-18 00:00:00
1,53cdb2fc8bc7dce0b6741e2150273451,b0830fb4747a6c6d20dea0b8c802d7ef,delivered,2018-07-24 20:41:37,2018-07-26 03:24:27,2018-07-26 14:31:00,2018-08-07 15:27:45,2018-08-13 00:00:00
2,47770eb9100c2d0c44946d9cf07ec65d,41ce2a54c0b03bf3443c3d931a367089,delivered,2018-08-08 08:38:49,2018-08-08 08:55:23,2018-08-08 13:50:00,2018-08-17 18:06:29,2018-09-04 00:00:00


## Geolocation Table
Latitude and longitude coordinates for Brazilian zip codes.
Used for mapping customer and seller locations.

- **Rows:** ~1,000,163 | **Columns:** 5
- **geolocation_zip_code_prefix:** links to customers and sellers via zip code
- **geolocation_lat / geolocation_lng:** coordinates for map plotting

**Why so many rows?** Multiple coordinates per zip code — Brazilian zip codes
cover areas, not single points. We'll use this in Month 2 for geographic visualisations.
Skip detailed analysis of this table for now.

In [12]:
print("=== TRANSLATIONS ===")
print("Shape:", translations.shape)          # rows and columns
print("\nColumn names:")
print(translations.columns.tolist())         # all column names
print("\nFirst 3 rows:")
orders.head(3)

=== TRANSLATIONS ===
Shape: (71, 2)

Column names:
['product_category_name', 'product_category_name_english']

First 3 rows:


,order_id,customer_id,order_status,order_purchase_timestamp,order_approved_at,order_delivered_carrier_date,order_delivered_customer_date,order_estimated_delivery_date
0,e481f51cbdc54678b7cc49136f2d6af7,9ef432eb6251297304e76186b10a928d,delivered,2017-10-02 10:56:33,2017-10-02 11:07:15,2017-10-04 19:55:00,2017-10-10 21:25:13,2017-10-18 00:00:00
1,53cdb2fc8bc7dce0b6741e2150273451,b0830fb4747a6c6d20dea0b8c802d7ef,delivered,2018-07-24 20:41:37,2018-07-26 03:24:27,2018-07-26 14:31:00,2018-08-07 15:27:45,2018-08-13 00:00:00
2,47770eb9100c2d0c44946d9cf07ec65d,41ce2a54c0b03bf3443c3d931a367089,delivered,2018-08-08 08:38:49,2018-08-08 08:55:23,2018-08-08 13:50:00,2018-08-17 18:06:29,2018-09-04 00:00:00


## Category Name Translations Table
Maps Portuguese product category names to English.
Small but essential — makes all category analysis readable.

- **Rows:** ~71 | **Columns:** 2
- **product_category_name:** Portuguese name (e.g. `cama_mesa_banho`)
- **product_category_name_english:** English name (e.g. `bed_bath_table`)

**How to use it:** Merge with the products table using `product_category_name`
as the join key. After merging, all category names become English automatically.
We will do this merge in the next notebook.

In [13]:
print("=== NULL VALUES PER TABLE ===\n")

tables = {
    "orders": orders,
    "order_items": order_items,
    "order_payments": order_payments,
    "order_reviews": order_reviews,
    "customers": customers,
    "sellers": sellers,
    "products": products
}

for name, df in tables.items():
    nulls = df.isnull().sum()
    nulls = nulls[nulls > 0]  # only show columns WITH nulls
    if len(nulls) > 0:
        print(f"--- {name} ---")
        print(nulls)
        print()

=== NULL VALUES PER TABLE ===

--- orders ---
order_approved_at                 160
order_delivered_carrier_date     1783
order_delivered_customer_date    2965
dtype: int64

--- order_reviews ---
review_comment_title      87656
review_comment_message    58247
dtype: int64

--- products ---
product_category_name         610
product_name_lenght           610
product_description_lenght    610
product_photos_qty            610
product_weight_g                2
product_length_cm               2
product_height_cm               2
product_width_cm                2
dtype: int64

